src/Storage/CloudinaryStorage.php

```diff
@@ -58,12 +59,19 @@ public function setFromLocalFile($path, $filename = null, $hash = null, $variant
     {
         $filename = str_replace('\\', '/', $filename);
         $parts = explode('/', $filename);
-        $publicId = explode('.', array_pop($parts));
+
+        $info = pathinfo($filename);
+        $publicId = $info['filename'];
+        $extension = $info['extension'];
 
         $options = [
             'public_id' => $publicId[0],
-            'folder' => implode('/', $parts),
+            'folder' => implode('/', $parts)
```

Comment:
Won't `$parts` still have the file name at this point??
That's one of the reasons for using `array_pop()`